pydeseq2/ds.py

```diff
@@ -166,16 +164,26 @@ def __init__(
         self.lfc_null = lfc_null
         self.alt_hypothesis = alt_hypothesis
 
-        # Check the validity of the contrast (if provided) or build it.
-        self._build_contrast(contrast)
-
         # Initialize the design matrix and LFCs. If the chosen reference level are the
         # same as in dds, keep them unchanged. Otherwise, change reference level.
         self.design_matrix = self.dds.obsm["design_matrix"].copy()
         self.LFC = self.dds.varm["LFC"].copy()
 
-        # Build a contrast vector corresponding to the variable and levels of interest
-        self._build_contrast_vector()
+        # Check the validity of the contrast (if provided) or build it.
+        if contrast is None:
+            raise ValueError(
+                """Default contrasts are no longer supported.
+                The "contrast" argument must be provided."""
+            )
+        elif isinstance(contrast, np.ndarray):
+            assert (
+                contrast.shape[0] == self.dds.obsm["design_matrix"].shape[1]
+            ), "The contrast vector must have the same length as the design matrix."
```

Comment:
IMO this should raise a ValueError. 

assertions are only meant for internal consistency checks, and might be skipped depending on the python optimization settings.